### PR TITLE
fix(sdk): convert platform endpoint to grpc dial format

### PR DESCRIPTION
**Pull request by jrschumacher**

### CODE REVIEW

File: sdk/sdk.go [hunk 1]
```diff
@@ -143,6 +152,34 @@ func New(platformEndpoint string, opts ...Option) (*SDK, error) {
...
+
+	if !regexp.MustCompile(`^(https?:\/\/)?(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(:\d+)?|(localhost)(:\d+)?)(\/.*)?$`).MatchString(e) {
+		return "", errors.New("platform endpoint is not a valid URL")
+	}
+
+	u, err := url.ParseRequestURI(e)
+	if err != nil {
+		return "", errors.Join(fmt.Errorf("cannot parse platform endpoint(%s)", e), err)
+	}
+	fmt.Printf("u: %v\n", u)
```
Comment (jrschumacher) [inline sdk/sdk.go:169]:
```suggestion
```

File: sdk/sdk_test.go [hunk 2]
```diff
@@ -125,3 +125,84 @@ func Test_ShouldCreateNewSDKWithBadEndpoint(t *testing.T) {
 	require.NoError(t, err)
 	assert.NotNil(t, s)
 }
+
+func Test_ShouldSantizePlatformEndpoint(t *testing.T) {
```
Comment (dmihalcik-virtru) [inline sdk/sdk_test.go:129]: 🎅🏻 🎅🏻 🎅🏻 🎅🏻

File: sdk/sdk.go (hunk 1 above)
Comment (dmihalcik-virtru) [inline sdk/sdk.go:171]:
```suggestion
```

File: sdk/sdk.go [hunk 3]
```diff
@@ -143,6 +152,34 @@ func New(platformEndpoint string, opts ...Option) (*SDK, error) {
 	}, nil
 }
 
+func SanitizePlatformEndpoint(e string) (string, error) {
+	// check if there's a scheme, if not, add https
+	if !regexp.MustCompile(`^https?://`).MatchString(e) {
+		e = "https://" + e
+	}
+
+	if !regexp.MustCompile(`^(https?:\/\/)?(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(:\d+)?|(localhost)(:\d+)?)(\/.*)?$`).MatchString(e) {
```
Comment (dmihalcik-virtru) [inline sdk/sdk.go:163]:
```suggestion
	if !regexp.MustCompile(`^(https?:\/\/)?(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(:\d+)?|(localhost)(:\d+)?)\/?$`).MatchString(e) {
```
Do we want to allow paths on the platform endpoints?

File: sdk/sdk.go [hunk 4]
```diff
@@ -143,6 +152,34 @@ func New(platformEndpoint string, opts ...Option) (*SDK, error) {
...
+		return "", errors.New("platform endpoint is not a valid URL")
```
Comment (dmihalcik-virtru) [inline sdk/sdk.go:164]:
```suggestion
		return "", errors.New("platform endpoint is not valid")
```

File: sdk/options.go [hunk 5]
```diff
@@ -151,3 +152,11 @@ func WithPlatformConfiguration(platformConfiguration PlatformConfiguration) Opti
 		c.platformConfiguration = platformConfiguration
 	}
 }
+
+// WithIPC returns an Option that indicates the SDK should use IPC for communication
+// this will allow the platform endpoint to be an empty string
+func WithIPC() Option {
```
Comment (dmihalcik-virtru) [inline sdk/options.go:158]: Suggestion - why not use a custom scheme, since this implies a different transport?

File: sdk/sdk.go (hunk 3 above)
Comment (jrschumacher) [reply sdk/sdk.go:163]: Probably not. Rather tell people they need to run this on the root of the domain.

File: sdk/options.go (hunk 5 above)
Comment (jrschumacher) [reply sdk/options.go:158]: The issue with this is that there is no scheme at all. When we bind the SDK into the platform, it's an empty string `sdk.New("")` I'm open to a new name... maybe `WithNoEndpoint()` or something like that?

Comment (dmihalcik-virtru) [reply sdk/options.go:158]: I'm fine with this since we intend to support it indefinitely

**Review by dmihalcik-virtru — DISMISSED** [review]
It looks like you need to run `make fix` to get some transitive updates.

**Review by strantalis — APPROVED** [review]

**Pull request merged**